### PR TITLE
Improving "internal reload" during project reload

### PR DESCRIPTION
**Pull request by sdedic**
Seems awkward... but sometimes events are delivered *in the middle* of a project loading process. The API checks the project's state at the end for quality and/or consistency - and that can cause spurious errors about project reloads while reading it. This PR improves stability by allowing a recovery - but at the same time if a reload would be repeated again because of file changes, the `ProjectReload.withProjectState()` fails indicating that project is being modified, as it was before. Similar approach was already implemented for the case the `ProjectReloadImplementation` requested a retry: one is OK, but the same (super)set repeated again will result in an error.

During investigation, I needed quite detailed logging, so I've added that as a separate commit; the "logging" commit should bring no functional changes, just diagnostics.

### CODE REVIEW

**Review by lahodaj — APPROVED** [review]
No objections from me.

File: ide/project.dependency/src/org/netbeans/modules/project/dependency/ProjectReload.java
Comment (mbien) [inline ide/project.dependency/src/org/netbeans/modules/project/dependency/ProjectReload.java:968]: guard with isLoggable like a few lines below. Since `toString()` of ProjectState is going to do some work now.

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java
Comment (mbien) [inline java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java:1795]: check return value and log if false?

File: ide/project.dependency/src/org/netbeans/modules/project/dependency/reload/Reloader.java
Comment (mbien) [inline ide/project.dependency/src/org/netbeans/modules/project/dependency/reload/Reloader.java:469]: if i see this correctly on gh, the inner guard is redundant

**Comment by sdedic** [comment]
Rebased on latest master; @mbien review comments addressed.

File: ide/project.dependency/src/org/netbeans/modules/project/dependency/reload/Reloader.java
Comment (sdedic) [reply ide/project.dependency/src/org/netbeans/modules/project/dependency/reload/Reloader.java]: addressed in 2d925f2

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java
Comment (sdedic) [reply java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImpl.java]: addressed in 2d925f2

File: ide/project.dependency/src/org/netbeans/modules/project/dependency/ProjectReload.java
Comment (sdedic) [reply ide/project.dependency/src/org/netbeans/modules/project/dependency/ProjectReload.java]: addressed in 2d925f2

**Comment by mbien** [comment]
> Rebased on latest master; @mbien review comments addressed.

ok thanks, don't forget to squash the third commit into the other commits before merge though.

**Comment by sdedic** [comment]
> ok thanks, don't forget to squash the third commit into the other commits before merge though.

Sure, waiting until Mon for other possible review comments + fixes.

**Review by jhorvath — APPROVED** [review]

**Review by MartinBalin — APPROVED** [review]

**Comment by sdedic** [comment]
Fixes merged into 2nd commit; prepare for merge.

**Pull request merged**
